lib/travis/api/attack.rb

```diff
@@ -35,6 +36,11 @@ def self.bantime(value)
     /\.(png|svg)$/.match(request.path)
   end
 
+  # https://help.github.com/articles/what-ip-addresses-does-github-use-that-i-should-whitelist/
+  whitelist('safelist anything coming from github') do |request|
+    request.ip && NetAddr::CIDR.create('192.30.252.0/22').contains?(request.ip)
```

Comment:
Looks great! Maybe we can look into memoizing this `CIDR` object so that we don't allocate a new object on every API request?